### ISSUE TITLE
Residual Attention Network for Image Classification

### ISSUE THREAD

**Issue by sooooner** [sooooner/papers#46]
# Residual Attention Network for Image Classification [[link](https://arxiv.org/abs/1704.06904)]

### Abstract
Inside each Attention Module, bottom-up top-down feedforward structure is used to unfold the feedforward and feedback attention process into a single feedforward process

## Introduction
attention은 forcucing location 뿐만 아니라, 해당 위치의 object의 다양한 representation을 향상 시킴
model의 특징
1. attention module의 수를 증가시키면 성능이 일정하게 향상됨 + 각 module은 서로 다은 형식의 attention을 감지하도록 학습됨
2. end-to-end

논문의 기여
1. 여러 attention을 쌓아서 서로 다른 attenttion을 감지하도록 함
2. 단순히 쌓는것 보다 residual 구조를 사용하여 성능 향상
3. bottom-up, top-down feed-forward attention을 사용하여 feature에 soft weight를 추가하는 용도로 사용함

![image](https://user-images.githubusercontent.com/31437443/114267123-02737f80-9a35-11eb-9e14-3bf9e97e2ecc.png)

## Residual Attention Network

![image](https://user-images.githubusercontent.com/31437443/114267317-bd038200-9a35-11eb-9ea7-5513c24426be.png)

각 attention module은 mask(M)와 trunk(T) branch로 나누어짐
+ feature processing으로 다른 NN구조에 적용이 가능함
+ T의 output T(x)가 주어기면 M은 T(x)에 soft weight를 주는 같은 size의 M(x)를 bottom-up, top-down structure로 학습함

![image](https://user-images.githubusercontent.com/31437443/114267143-1dde8a80-9a35-11eb-8356-6f8dc6313dcc.png)

attention Mask는 피드포워드 동안 feature selector + backprop 동안 gradient update filter 역할을 함

여러 module을 쌓는 이유
1. 복잡한 장면, 많은 모양 변화를 가지는 경우, 하나의 모듈로는 한계가 있음
2.  단일 모듈은 feature를 한 번만 수정함. 이미지의 일부에서 수정이 실패하면 다음 네트워크 모듈은 두 번째 기회가 없음

![image](https://user-images.githubusercontent.com/31437443/114267602-a6f6c100-9a37-11eb-8829-60b29de0ef3a.png)

### Attention Residual Learning
단순히 Att를 쌓기만 하면 성능 저하가 일어남
1. 0~1인 Mask를 계속 적용하면 feature가 약해짐
2. Mask가 Trunk의 residual unit의 identical mapping성질을 깨버림

위와 같은 이유로 Attention Residual Learning을 제안함
![image](https://user-images.githubusercontent.com/31437443/114267368-166bb100-9a36-11eb-86c1-ea2075bf7dca.png)

ResNet에서 F는 잔차, 논문의 F는 DCNN으로 생성됨 feature

M(x)는 좋은 feature를 강화하는 feature selector와 Trunk feature로 부터 온 noise를 억제하는 일을함

Att Res Learning은 original feature의 좋은 속성을 유지할 수 있지만 soft Mask Branch를 우회해 상위 레이어로 이동 하며 M의 faeture selection 기능을 제공 할 수도 있다.

### Soft Mask Branch
DBN의 attention mechanism은 
1. 전체 이미지의 글로벌 정보를 빠르게 수집함
2. 글로벌 정보를 원본 feature map과 결합함

CNN의 attention mechanism은 위 1, 2단계를 bottom-up, top-down fully conv structure로 전개됨

input에서 maxpooling을 여러번 수행하여 small # res unit의 receptive field를 빠르게 증가 시킴
가장 낮은 해상도에 도달 한 후 글로벌 정보는 symmetrical Top-down architecture로 확장되어 각 위치의 input feature로 들어감
Res unit을 몇개 통과한 후 linear interpolation으로 출력을 샘플링함
interpolation 이후 1 x 1 conv 두개를 거쳐 sigmoid로 정규화 또한 maxpool, interporation 전후로 skip connection을 두어서 다양한 스케일에서의 정보를 캡쳐함

![image](https://user-images.githubusercontent.com/31437443/114267499-fa1c4400-9a36-11eb-9313-c002964e4236.png)

### Spatial Attention and Channel Attention
mask 끝에 activation을 변형하여 정규화를 추가 하기로 함

![image](https://user-images.githubusercontent.com/31437443/114267546-44052a00-9a37-11eb-99f9-eff19772e067.png)

(4) : Mixed attention : sigmoid
(5) : channel attention : channel wise L2 norm : remove spatial information
(6) : spatial attention : feature map norm + sigmoid : get spatial information only

성능으 (4)번이 젤 좋음
